### ISSUE TITLE
Display better diffs for JSON literal properties

### ISSUE THREAD

**Issue by jen20** [pulumi/pulumi#5831]
We could do a better job of displaying diffs in (for example) IAM policies which are JSON strings.

```typescript
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:dev::iamdiff::pulumi:pulumi:Stack::iamdiff-dev]
    ~ aws:iam/policy:Policy: (update)
        [id=arn:aws:iam::153052954103:policy/testpolicy-8b49b63]
        [urn=urn:pulumi:dev::iamdiff::aws:iam/policy:Policy::testpolicy]
      ~ policy: "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Effect\":\"Allow\",\"Action\":[\"s3:CreateBucket\"],\"Resource\":[\"*\"]}]}" => "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Effect\":\"Allow\",\"Action\":[\"s3:CreateBucket\",\"s3:DeleteBucket\"],\"Resource\":[\"*\"]}]}"
    --aws:iam/rolePolicyAttachment:RolePolicyAttachment: (delete-replaced)
        [id=testrole-d8a65d0-20190522144955669700000001]
        [urn=urn:pulumi:dev::iamdiff::aws:iam/rolePolicyAttachment:RolePolicyAttachment::attach]
    +-aws:iam/rolePolicyAttachment:RolePolicyAttachment: (replace)
        [id=testrole-d8a65d0-20190522144955669700000001]
        [urn=urn:pulumi:dev::iamdiff::aws:iam/rolePolicyAttachment:RolePolicyAttachment::attach]
      ~ policyArn: "arn:aws:iam::153052954103:policy/testpolicy-8b49b63" => output<string>
    ++aws:iam/rolePolicyAttachment:RolePolicyAttachment: (create-replacement)
        [id=testrole-d8a65d0-20190522144955669700000001]
        [urn=urn:pulumi:dev::iamdiff::aws:iam/rolePolicyAttachment:RolePolicyAttachment::attach]
      ~ policyArn: "arn:aws:iam::153052954103:policy/testpolicy-8b49b63" => output<string>
```

We could consider use of a JSON patch library to show the actual diff here instead of a complete string, if we have some way of marking that a property is a JSON object.

**Comment by manospasj** [comment]
Any update on this? This will be very helpful for me as I am managing quite a few resources that produce JSON strings. I would also be happy to implement this if I can get some guidance

**Comment by Shrooblord** [comment]
@manospasj just in case you hadn't seen this -- it's been released in 3.29.1
